packages/vite-plugin-cloudflare/src/node-js-compat.ts

```diff
@@ -130,3 +142,45 @@ function getNodeCompatEntries() {
 
 	return entries;
 }
+
+export class NodeJsCompatWarnings {
+	private sources = new Map<string, Set<string>>();
+	private timer: NodeJS.Timeout | undefined;
+
+	constructor(private readonly environment: vite.Environment) {}
+
+	registerImport(source: string, importer = "<unknown>") {
+		const importers = this.sources.get(source) ?? new Set();
+		this.sources.set(source, importers);
+		if (!importers.has(importer)) {
+			importers.add(importer);
+		}
```

Comment:
is checking that the importer is not there necessary? does it make a difference (`importers` is a set so this would not cause any duplicated entries anyways)